#! /bin/sh
patch -p1 -f $* < $0
exit $?

commit b11b57fb1f8c5c31818691eac9a26074a0d89f8c
Author: Andrej Kacian <ticho@claws-mail.org>
Date:   Sun Jan 20 14:45:08 2019 +0100

    Fix session proxy initialization for Windows
    
    Fixes bug #4145: proxy server for sending doesn't work

diff --git a/src/common/session.c b/src/common/session.c
index 2dc0a0a91..a6f19ea80 100644
--- a/src/common/session.c
+++ b/src/common/session.c
@@ -122,7 +122,6 @@ void session_init(Session *session, const void *prefs_account, gboolean is_smtp)
  */
 gint session_connect(Session *session, const gchar *server, gushort port)
 {
-#ifdef G_OS_UNIX
 	session->server = g_strdup(server);
 	session->port = port;
 
@@ -131,6 +130,7 @@ gint session_connect(Session *session, const gchar *server, gushort port)
 		port = session->proxy_info->proxy_port;
 	}
 
+#ifdef G_OS_UNIX
 	session->conn_id = sock_connect_async(server, port, session_connect_cb,
 					      session);
 	if (session->conn_id < 0) {
@@ -145,9 +145,6 @@ gint session_connect(Session *session, const gchar *server, gushort port)
 #else
 	SockInfo *sock;
 
-	session->server = g_strdup(server);
-	session->port = port;
-
 	sock = sock_connect(server, port);
 	if (sock == NULL) {
 		g_warning("can't connect to server.");
